R/check_structure.R

```diff
@@ -96,6 +96,14 @@ check_structure <- function(protocol_code, fail = !interactive()) {
 
 
 check_file <- function(filename, x, files_template, path_to_template) {
+  # check if file is present in template
+  if (!filename %in% c(files_template, "index.Rmd")) {
+    x$add_error(
```

Comment:
This will check all ".Rmd" files, but it is allowed to have ".Rmd" files that are among the template .Rmd files. You can see examples [here](https://github.com/inbo/protocolsource/tree/sfp-401-nl/src/thematic/4_vegetation/sfp-401-nl_vegopname-terrest), where you have `_childdoc_visualisatie_bedekkingen.Rmd`. So I think you will also need to change line 51:

```
protocol_rmds <- files_protocol[grepl(".Rmd$", files_protocol)]
```

to restrict it to Rmd files that start with a chapter number?

Reply:
Ah, this is allowed.  Sorry, I didn't get that from the previous discussions.  Are there any naming conventions for these files that are not tested elsewhere?  E.g. filename has to start with an underscore?

Reply:
To be exact, we make use of [`rmd_files`](https://bookdown.org/yihui/bookdown/usage.html#:~:text=You%20can%20override,for%20the%20book) key in the template `_bookdown.yml`. So, only files that are listed in `rmd_files` will be merged by bookdown before knitting. This list of files is programmatically created by `create_protocol()`. So a user can in principle add any Rmd file (with any name), but it will simply be ignored when rendering unless (i) it is manually added to the `rmd_files` list, or (ii) it is referred to in an R chunk as a child document.

Reply:
So files without chapter number shouldn't be checked at all, also not the headings (layout, presence of 1st level headings,...) and the correctness of chunks?

Also, if these files are ignored, you may want to warn the user about that if it is not clearly indicated in the manual.  And maybe you also may want to have some kind of warning to be able to check if this is a relevant file, or an old version of a file that is left there by the author and that actually has to be removed.  But I don't know which is the best way to handle it: it seems good to have the warning at least once, but one should be able to mute it...  We could solve it similarly to checklist, by saving somewhere a list with 'accepted' files and show all the others as a warning together with the errors.

These are just ideas, up to you to decide what is useful or necessary.  Keep in mind you don't want to check the protocols manually, but on the other hand you also don't want to be bothered by useless checks.

Reply:
> So files without chapter number shouldn't be checked at all, also not the headings (layout, presence of 1st level headings,...) and the correctness of chunks?

Indeed I think this is not necessary. For instance, a child document does not have to have headings. If something is wrong with chunks, at least an error will be returned.

> Also, if these files are ignored, you may want to warn the user about that if it is not clearly indicated in the manual. And maybe you also may want to have some kind of warning to be able to check if this is a relevant file, or an old version of a file that is left there by the author and that actually has to be removed. But I don't know which is the best way to handle it: it seems good to have the warning at least once, but one should be able to mute it... We could solve it similarly to checklist, by saving somewhere a list with 'accepted' files and show all the others as a warning together with the errors.

I agree these are good ideas, but not high priority and the exact details of implementation will need to wait. Can you maybe write an issue so we can keep track of these ideas? In any case postpone work for later, getting the website up and running is higher priority.

Reply:
> Indeed I think this is not necessary. For instance, a child document does not have to have headings. If something is wrong with chunks, at least an error will be returned.

It indeed doesn't have to have headings, but is it for instance a problem if it has first level headings?  It is not a big deal to add (a selection of) the existing checks for the chapters to other `.Rmd` documents as well, I just have to know your choice on it.

The other idea is added to an issue.

Reply:
Indeed, here are some relevant, nice to have, checks for the other Rmd files:

- should not have level 1 header
- general header checks (same as those in check_file, i.e. check if ## is not proceeded by space and no whitespace at the end of heading)
- general chunk checks (same as those in check_file)

Reply:
> Indeed, here are some relevant, nice to have, checks for the other Rmd files:
> 
> * should not have level 1 header
> * general header checks (same as those in check_file, i.e. check if ## is not proceeded by space and no whitespace at the end of heading)
> * general chunk checks (same as those in check_file)

Eh, what are you trying to say?  Is it useful to apply these checks to the `.Rmd` files without chapter number as well, or not?

Reply:
Yes, this is useful